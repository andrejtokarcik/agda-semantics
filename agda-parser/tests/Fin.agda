module Fin where

data Nat : Set where
  zero : Nat
  succ : Nat -> Nat

data Fin : Nat -> Set where
    fzero : {n : Nat} -> Fin (succ n)
    fsucc : {n : Nat} -> Fin n -> Fin (succ n)
