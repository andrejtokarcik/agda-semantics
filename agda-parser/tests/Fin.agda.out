<k>
    whnf ( Fun ( "t1" ) ) ~> nf ( HOLE )
</k>
<dbg>
    0
</dbg>
<mgu>
    subst(.KList)
</mgu>
<ctxstack>
    .List
</ctxstack>
<sig>
    Data ( "Fin" ) |-> ( #symVariable(5) : Data ( "Nat" ) ) -> Set ( 0 )
    Data ( "Nat" ) |-> Set ( 0 )
    Con ( "fsucc" ) |-> { #symVariable(28) : Data ( "Nat" ) } -> ( 
      #symVariable(27) : (Data ( "Fin" ) #symVariable(28)) ) -> (Data ( "Fin" 
      ) (Con ( "succ" ) #symVariable(28)))
    Con ( "fzero" ) |-> { #symVariable(14) : Data ( "Nat" ) } -> (Data ( "Fin"
       ) (Con ( "succ" ) #symVariable(14)))
    Con ( "succ" ) |-> ( #symVariable(4) : Data ( "Nat" ) ) -> Data ( "Nat" )
    Con ( "zero" ) |-> Data ( "Nat" )
</sig>
<ctx>
    .Map
</ctx>
<meta>
    .Map
</meta>
<ncat>
    Name ( "Fin" ) |-> Data ( "Fin" )
    Name ( "Nat" ) |-> Data ( "Nat" )
    Name ( "fsucc" ) |-> Con ( "fsucc" )
    Name ( "fzero" ) |-> Con ( "fzero" )
    Name ( "succ" ) |-> Con ( "succ" )
    Name ( "zero" ) |-> Con ( "zero" )
</ncat>
<fun>
    .Map
</fun>
